scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -284,6 +288,34 @@ object Execution {
   }
 
   /**
+    * This is a Thread used as a shutdown hook to clean up temporary files created by some Execution
+    */
+  case class TempFileCleanup(filesToCleanup: mutable.Set[String], mode: Mode) extends Thread {
```

Comment:
maybe also call out that if the job is aborted this will not cleanup stuff (if folks haven't read / know about shutdown hook semantics)